File: backend/src/main/java/com/healthsphere/backend/service/PrescriptionService.java
package com.healthsphere.backend.service;

import com.healthsphere.backend.dto.PrescriptionRequest;
import com.healthsphere.backend.dto.PrescriptionResponse;

import java.util.List;

public interface PrescriptionService {

    PrescriptionResponse createPrescription(PrescriptionRequest request);

    List<PrescriptionResponse> getAllPrescriptions();

    PrescriptionResponse getPrescriptionById(Long id);

    PrescriptionResponse updatePrescription(Long id, PrescriptionRequest request);

    void deletePrescription(Long id);
}

File: frontend/src/services/prescriptionService.js
import api from "./api";

const URL = "/prescriptions";

const prescriptionService = {

  getAllPrescriptions: async () => {
    const response = await api.get(URL);
    return response.data;
  },

  getPrescriptionById: async (id) => {
    const response = await api.get(`${URL}/${id}`);
    return response.data;
  },

  createPrescription: async (prescription) => {
    const response = await api.post(URL, prescription);
    return response.data;
  },

  updatePrescription: async (id, prescription) => {
    const response = await api.put(`${URL}/${id}`, prescription);
    return response.data;
  },

  deletePrescription: async (id) => {
    const response = await api.delete(`${URL}/${id}`);
    return response.data;
  },

};

export default prescriptionService;